JASP-Engine/JASP/R/classicalmetaanalysis.R

```diff
@@ -124,39 +124,49 @@ ClassicalMetaAnalysis <- function(dataset=NULL, options, perform="run", callback
     )
     
     .hasErrors(dataset, perform, type=c("infinity", "observations"),
-      all.target=numeric.variables, observations.amount="< 2",
-      exitAnalysisIfErrors=TRUE)
+               all.target=numeric.variables, observations.amount="< 2",
+               exitAnalysisIfErrors=TRUE)
   }
   
   method = switch(options$method, `Fixed Effects` = "FE", `Maximum Likelihood` = "ML",
-      `Restricted ML` = "REML", `DerSimonian-Laird` = "DL", `Hedges` = "HE",
-      `Hunter-Schmidt` = "HS", `Sidik-Jonkman` = "SJ", `Empirical Bayes` = "EB",
-      `Paule-Mandel` = "PM", "REML")
-
+                  `Restricted ML` = "REML", `DerSimonian-Laird` = "DL", `Hedges` = "HE",
+                  `Hunter-Schmidt` = "HS", `Sidik-Jonkman` = "SJ", `Empirical Bayes` = "EB",
+                  `Paule-Mandel` = "PM", "REML")
+  
   can.run = all(c(effsizeName, stderrName) != "")
   
   ## Run the analysis. ####
   if (is.null(rma.fit)) {
-    rma.fit <- structure(list('b' = numeric(),'se' = numeric(),'ci.lb' = numeric(),'ci.ub' = numeric(), 
+    rma.fit <- structure(list('b' = numeric(),'se' = numeric(),'ci.lb' = numeric(), 'ci.ub' = numeric(), 
                     'zval' = numeric(),'pval' = numeric()), class = c("dummy", "rma"))
     
     if (run && can.run) {
       
+      # infer model formula
       .vmodelTerms = options$modelTerms
       formula.rhs <- as.formula(as.modelTerms(.vmodelTerms))
       if (is.null(formula.rhs)) 
         formula.rhs = ~1
+      
       if (!options$includeConstant) 
         formula.rhs = update(formula.rhs, ~ . + 0)
-      if( identical(formula.rhs, ~ 1 - 1) )
+      
+      if (identical(formula.rhs, ~ 1 - 1))
         .quitAnalysis("The model should contain at least one predictor or an intercept.")
-
-      rma.fit <- metafor::rma(
+      
+      # check input data
+      if (any(dataset[[b64(stderrName)]] <= 0))
```

Comment:
You can add this to the .hasErrors call. This has the added benefit of grouping all error checking and the ability to show multiple failures in one message.

```
    # check input data
    checkSE <- function() {
      if (any(dataset[[b64(stderrName)]] <= 0))
        paste0("Non-positive standard errors encountered in variable '", stderrName, "'. Standard errors should be positive. Please check your input.")
    }
    
    .hasErrors(dataset, perform, type=c("infinity", "observations"), custom=checkSE,
               all.target=numeric.variables, observations.amount="< 2",
               exitAnalysisIfErrors=TRUE)

```